src/realm/deppart/byfield_gpu_impl.hpp

```diff
@@ -0,0 +1,261 @@
+#pragma once
+#include "realm/deppart/byfield.h"
```

Comment:
I've left comments on this file with mostly stylistic things. I'm not yet going to read the other parts of the pipeline, because they share the same problem as here: there are not enough comments or suggestive variable names that describe the overall picture of what this function is actually computing, and how the pieces fit together. Right now, it's not possible to follow the code as someone who didn't write it. This is a pretty algorithmic piece of code -- you should be able to describe in comments (maybe with small examples or descriptions or textual diagrams) about what each phase of the computation is doing, and how that fits into the overall algorithm.